openedx/core/djangoapps/programs/tasks/v1/tasks.py

```diff
@@ -194,14 +194,9 @@ def award_program_certificates(self, username):
                 )
             except Exception:  # pylint: disable=broad-except
                 # keep trying to award other certs, but retry the whole task to fix any missing entries
-                warning_message = 'Failed to award certificate for program {uuid} to user {username}.'.format(
-                    uuid=program_uuid, username=username)
-
-                if countdown < MAX_RETRIES:
-                    LOGGER.warning(warning_message)
-                    retry = True
-                else:
-                    LOGGER.exception("Max retries exceeded. {msg}".format(msg=warning_message))
+                LOGGER.warning('Failed to award certificate for program {uuid} to user {username}.'.format(
```

Comment:
I think that this looks good.  Can we also add an additional warning if this is the last attempt

```
if self.request.retries == MAX_RETRIES:
    LOGGER.exception('Final attempt to aware certificate for program %s to user %s failed', program_uuid, username)
```

Reply:
::EDIT::

Actually I think this may not be necessary as you mentioned in the other PR.  But we should update our alerting to capture whatever error message is output when MaxRetriesExceededError is thrown

Reply:
https://github.com/celery/celery/blob/d02d260a192f8923dfa331ae2b503af57534654a/celery/app/task.py#L673

Reply:
@bderusha Got it, thanks.